lib/compat/wordpress-6.5/fonts/fonts.php

```diff
@@ -84,6 +84,35 @@ function gutenberg_create_initial_post_types() {
 	);
 }
 
+/**
+ * Filters the user capabilities to grant the 'upload_fonts' capability as necessary.
+ *
+ * To grant the 'upload_fonts' capability, file modifications must be allowed, the fonts directory must be
+ * writable, and the user must have the 'edit_theme_options' capability.
+ *
+ * @since 6.5.0
+ *
+ * @param bool[] $allcaps An array of all the user's capabilities.
+ * @return bool[] Filtered array of the user's capabilities.
+ */
+function gutenberg_maybe_grant_upload_font_cap( $allcaps, $caps ) {
+	if ( ! in_array( 'upload_fonts', $caps, true ) ) {
+		return $allcaps;
+	}
+
+	$fonts_dir = wp_get_font_dir()['path'];
+	if (
+		wp_is_file_mod_allowed( 'can_upload_fonts' ) &&
+		wp_is_writable( $fonts_dir ) &&
+		! empty( $allcaps['edit_theme_options'] )
```

Comment:
Developers can modify the cap required for editing the font face/families (similar to the example in https://github.com/WordPress/gutenberg/pull/59104#discussion_r1496613114 ) so I am not sure this is required. 

`canUser( 'create' ... )` should catch this.

However, I'm not entirely sure it is not required, either, so will require a logic check on this.